src/Avalonia.Controls/ToolTipService.cs

```diff
@@ -122,30 +121,6 @@ private void TipChanged(AvaloniaPropertyChangedEventArgs e)
             }
         }
 
-        private void TipOpenChanged(AvaloniaPropertyChangedEventArgs e)
-        {
-            var control = (Control)e.Sender;
-
-            if (e.OldValue is false && e.NewValue is true)
-            {
-                control.DetachedFromVisualTree += ControlDetaching;
-                control.EffectiveViewportChanged += ControlEffectiveViewportChanged;
-            }
-            else if (e.OldValue is true && e.NewValue is false)
-            {
-                control.DetachedFromVisualTree -= ControlDetaching;
-                control.EffectiveViewportChanged -= ControlEffectiveViewportChanged;
-            }
-        }
-
-        private void ControlDetaching(object? sender, VisualTreeAttachmentEventArgs e)
-        {
-            var control = (Control)sender!;
-            control.DetachedFromVisualTree -= ControlDetaching;
-            control.EffectiveViewportChanged -= ControlEffectiveViewportChanged;
-            Close(control);
```

Comment:
TODO: Need to handle situation when Popup is closed by itself (target detached...), without resetting ToolTip.IsOpen value.

Reply:
Related tests are failing as well